src/cmd/linuxkit/run_qemu.go

```diff
@@ -525,13 +526,19 @@ func buildQemuCmdline(config QemuConfig) (QemuConfig, []string) {
 	if config.NetdevConfig == "" {
 		qemuArgs = append(qemuArgs, "-net", "none")
 	} else {
-		mac := generateMAC()
-		qemuArgs = append(qemuArgs, "-net", "nic,model=virtio,macaddr="+mac.String())
+		var macRemoveRegExp = regexp.MustCompile(",((?:[0-9A-Fa-f]{2}[:-]){5}[0-9A-Fa-f]{2})|((?:[0-9A-Fa-f]{2}[:-]){5}[0-9A-Fa-f]{2}),")
+		var macRegExp = regexp.MustCompile("([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})")
+		var mac = macRegExp.FindString(config.NetdevConfig)
```

Comment:
There seems to be a function in the go library to parse a MAC address: https://golang.org/pkg/net/#ParseMAC

Any reason you are not using it?